custom_components/vicare/binary_sensor.py

```diff
@@ -0,0 +1,187 @@
+
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+import requests
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_POWER,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG, VICARE_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_CIRCULATION_PUMP_ACTIVE = "circulationpump_active"
+SENSOR_BURNER_ACTIVE = "burner_active"
+SENSOR_COMPRESSOR_ACTIVE = "compressor_active"
+
+
+@dataclass
+class ViCareBinarySensorEntityDescription(
+    BinarySensorEntityDescription, ViCareRequiredKeysMixin
+):
+    """Describes ViCare binary sensor entity."""
+
+
+GLOBAL_SENSORS: tuple[ViCareBinarySensorEntityDescription, ...] = (
+    ViCareBinarySensorEntityDescription(
+        key=SENSOR_CIRCULATION_PUMP_ACTIVE,
+        name="Circulation pump active",
+        device_class=DEVICE_CLASS_POWER,
+        value_getter=lambda api: api.getCirculationPumpActive(),
+    ),
+)
+
+BURNER_SENSORS: tuple[ViCareBinarySensorEntityDescription, ...] = (
+    ViCareBinarySensorEntityDescription(
+        key=SENSOR_BURNER_ACTIVE,
+        name="Burner active",
+        device_class=DEVICE_CLASS_POWER,
+        value_getter=lambda api: api.getActive(),
+    ),
+)
+
+CIRCUIT_SENSORS: tuple[ViCareBinarySensorEntityDescription, ...] = (
+    ViCareBinarySensorEntityDescription(
+        key=SENSOR_COMPRESSOR_ACTIVE,
+        name="Compressor active",
+        device_class=DEVICE_CLASS_POWER,
+        value_getter=lambda api: api.getCompressorActive(),
+    ),
+)
+
+
+def _build_entity(name, vicare_api, device_config, sensor):
+    try:
+        sensor.value_getter(vicare_api)
+        _LOGGER.debug("Found entity %s", name)
+        return ViCareBinarySensor(
+            name,
+            vicare_api,
+            device_config,
+            sensor,
+        )
+    except PyViCareNotSupportedFeatureError:
+        _LOGGER.info("Feature not supported %s", name)
+        return None
+    except AttributeError:
+        _LOGGER.debug("Attribute Error %s", name)
+        return None
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Create the ViCare binary sensor devices."""
+    name = hass.data[DOMAIN][config_entry.entry_id][VICARE_NAME]
+    api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
+
+    all_devices = []
+    for description in GLOBAL_SENSORS:
+        entity = _build_entity(
+            f"{name} {description.name}",
+            hass.data[DOMAIN][config_entry.entry_id][VICARE_API],
+            hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG],
+            description,
+        )
+        if entity is not None:
+            all_devices.append(entity)
+
+    for description in CIRCUIT_SENSORS:
```

Comment:
Here should be iterated over the compressors + compressor senors and not circuits, just like burners with try / catch because they are optional.
```suggestion
    for description in COMPRESSOR_SENSOR:
```